do/droplets.go

```diff
@@ -200,8 +200,11 @@ func dropletByName(ctx context.Context, client *godo.Client, nodeName types.Node
 	}
 
 	for _, droplet := range droplets {
-		if droplet.Name == string(nodeName) {
```

Comment:
I think this check should still be in place. I see that `nodeAddresses` would still catch this cause `Hostname` address type would still match `nodeName`, but it's  explicit and might be hard to debug later. WDYT?

Reply:
If people is debugging they gonna look at `nodeAddresses` anyway, and I think it is obvious enough that it also handle `nodeHostname`, but that maybe just me?

You decide :)

Reply:
I think it's very obvious to you and me, but it may not always be obvious that the node hostname = droplet name and that the hostname is an address type. I think we should keep the `droplet.Name == string(nodeName)` check :)

Reply:
Fixed..